app/Commands/Know/Search.php

```
@@ -0,0 +1,212 @@
+        try {
+            $remoteUrl = $this->runGitCommand(['git', 'remote', 'get-url', 'origin']);
+            if ($remoteUrl && preg_match('#github\.com[/:]([^/]+/[^/]+?)(?:\.git)?/?$#', $remoteUrl, $matches)) {
+                $context['repo'] = $matches[1];
+            }
+            $context['branch'] = $this->runGitCommand(['git', 'branch', '--show-current']);
+        } catch (\Exception $e) {
+            // Context is optional
+        }
+
+        return $context;
+    }
```

PHPMD (2.15.0)

Avoid unused private methods such as 'getGitContext'. (Unused Code Rules)

(UnusedPrivateMethod)